src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/InstantiateArgumentExceptionsCorrectly.cs

```diff
@@ -103,7 +105,7 @@ public override void Initialize(AnalysisContext context)
             }
             else
             {
-                Diagnostic? diagnostic = null;
+                List<Diagnostic> diagnostics = new();
```

Comment:
I don't think we need to use a collection
```suggestion
                Diagnostic? diagnosticFound = null;
```